### PR TITLE
Shrink dock panel tabs proportionally in case of overflow

### PR DESCRIPTION
**Pull request by cbjeukendrup**
Resolves: #9262
Closes: #10059

Schermafbeelding 2022-02-21 om 13 57 07 Schermafbeelding 2022-02-21 om 13 57 40

I experimented with this months ago, but couldn't get it to work properly. Instead, I created #10059 as an alternative solution.
However, after rebasing my old branch this morning, it magically started to work. So, I'm creating a PR for this solution too.

### CODE REVIEW

File: thirdparty/KDDockWidgets/src/private/quick/TabBarQuick.cpp [hunk 1]
```diff
@@ -124,7 +124,7 @@ QQuickItem *TabBarQuick::listView() const
 
     const QList<QQuickItem *> children = m_tabBarQmlItem->childItems();
     for (QQuickItem *child : children) {
-        if (qstrcmp(child->metaObject()->className(), "QQuickListView") == 0)
+        if (QString::fromUtf8(child->metaObject()->className()).startsWith(QString::fromUtf8("QQuickListView")))
```
Comment (cbjeukendrup) [inline thirdparty/KDDockWidgets/src/private/quick/TabBarQuick.cpp:127]: This is quite awful and hacky, but maybe the only way...

Slightly more beautiful would have been this: 
```c++
    const QMetaObject *listViewMetaObject = QMetaType::metaObjectForType(QMetaType::type("QQuickListView"));
    Q_ASSERT(listViewMetaObject);

    const QList<QQuickItem *> children = m_tabBarQmlItem->childItems();
    for (QQuickItem *child : children) {
        if (child->metaObject()->inherits(listViewMetaObject))
            return child;
    }
```
but that does not work (`listViewMetaObject` is nullptr).

**Review by RomanPudashkin — APPROVED** [review]

**Comment by DmitryArefiev** [comment]
Tested #9262 on Win10, Mac12, LinuxMint 20.2 - PASS      

Works this way:


https://user-images.githubusercontent.com/90187801/155991220-07a51ed0-de73-4fd7-b211-ead57d0c81f5.mp4



Please merge into master

**Comment by DmitryArefiev** [comment]
@cbjeukendrup Oh I see it was my mistake.. I took builds from #10059

In builds from #10626 it looks as expected. 
But I noticed one thing -the panel tabs are open not in shrink style after reopening the score:


https://user-images.githubusercontent.com/90187801/155998473-5ecb0ffe-b11d-478f-b14b-1d75b7223424.mp4

**Comment by Tantacrul** [comment]
Excellent stuff guys!

**Pull request merged**